modules/k8s-operator-crd-support/main.tf

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cluster_endpoint         = "https://${var.cluster_endpoint}"
+  token                    = data.google_client_config.default.access_token
+  cluster_ca_certificate   = data.google_container_cluster.primary.master_auth.0.cluster_ca_certificate
+  private_key              = var.create_ssh_key && var.ssh_auth_key == null ? tls_private_key.k8sop_creds[0].private_key_pem : file(var.ssh_auth_key)
+  k8sop_creds_secret_key   = var.secret_type == "cookiefile" ? "cookie_file" : var.secret_type
+  should_download_manifest = var.operator_path == null ? true : false
+  manifest_path            = local.should_download_manifest ? "${path.root}/.terraform/tmp/config-management-operator.yaml" : var.operator_path
+}
+
+
+data "google_container_cluster" "primary" {
+  name     = var.cluster_name
+  project  = var.project_id
+  location = var.location
+}
+
+data "google_client_config" "default" {
+}
+
+module "k8sop_manifest" {
+  source  = "terraform-google-modules/gcloud/google"
+  version = "~> 0.5"
+  enabled = local.should_download_manifest
+
+  create_cmd_entrypoint  = "gsutil"
+  create_cmd_body        = "cp ${var.operator_latest_manifest_url} ${local.manifest_path}"
+  destroy_cmd_entrypoint = "rm"
+  destroy_cmd_body       = "-f ${local.manifest_path}"
+}
+
+
+module "k8s_operator" {
+  source                = "terraform-google-modules/gcloud/google"
```

Comment:
nit: I know we havent enabled skip_download before, but imho we should be allowing users to control skip_download for the gcloud module,[ just like in project factory](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/ba4c889bf2358d4fc481ca1573dcf4731ae233e8/variables.tf#L206)

Reply:
oh, this is great -- happy to add this here or in a fast following PR. wish i knew this was available, working on this was very slow because of the download and local install everytime!

Reply:
yup, I gave config sync a spin locally and manually set it to true for faster apply

Reply:
cool, added in 418f9ae